sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -247,6 +268,132 @@ def test_lasso_cv_positive_constraint():
+        StandardScaler(),
+        LinearModel(normalize=False, fit_intercept=True, **params)
+    )
+
+    is_multitask = clf_norm._get_tags().get("multioutput_only", False)
+
+    # prepare the data
+    n_samples, n_features = 100, 2
+    rng = np.random.RandomState(0)
+    w = rng.randn(n_features)
+    X = rng.randn(n_samples, n_features)
+    X += 20  # make features non-zero mean
+    y = X.dot(w)
+
+    # make classes out of regression
+    if is_classifier(LinearModel):
+        y[y > np.mean(y)] = -1
+        y[y > 0] = 1
+    if is_multitask:
+        y = np.stack((y, y), axis=1)
+
+    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)
+
+    if 'alpha' in params:
+        clf_norm.alpha = params['alpha']
+        if isinstance(clf_norm, (Lasso, LassoLars, MultiTaskLasso)):
+            clf_pipe[1].alpha = params['alpha'] * np.sqrt(X_train.shape[0])
+        if isinstance(clf_norm, (Ridge, RidgeClassifier)):
+            clf_pipe[1].alpha = params['alpha'] * X_train.shape[0]
+    if ((isinstance(clf_norm, (ElasticNet)) and
+         not isinstance(clf_norm, Lasso)) or
+        (isinstance(clf_norm, (MultiTaskElasticNet)) and
+         not isinstance(clf_norm, MultiTaskLasso))):
+        if params['l1_ratio'] == 1:
+            clf_pipe[1].alpha = params['alpha'] * np.sqrt(X_train.shape[0])
+        if params['l1_ratio'] == 0:
+            clf_pipe[1].alpha = params['alpha'] * X_train.shape[0]
+
+    clf_norm.fit(X_train, y_train)
+    y_pred_normalize = clf_norm.predict(X_test)
+
+    clf_pipe.fit(X_train, y_train)
+    y_pred_standardize = clf_pipe.predict(X_test)
+
+    assert_allclose(clf_norm.coef_ * clf_pipe[0].scale_, clf_pipe[1].coef_)
+    assert clf_pipe[1].intercept_ == pytest.approx(y_train.mean())
+    assert (clf_norm.intercept_ ==
+            pytest.approx(y_train.mean() -
+                          clf_norm.coef_.dot(X_train.mean(0))))
+    assert_allclose(y_pred_normalize, y_pred_standardize)
+
+
+@pytest.mark.parametrize(
+    "LinearModel, params",
+    [
+     (Lasso, {"tol": 1e-16, "alpha": 0.1}),
+     (LassoCV, {"tol": 1e-16}),
+     (ElasticNetCV, {}),
+     (RidgeClassifier, {"solver": 'sparse_cg', "alpha": 0.1}),
+     (ElasticNet, {"tol": 1e-16, 'l1_ratio': 1, "alpha": 0.1}),
+     (ElasticNet, {"tol": 1e-16, 'l1_ratio': 0, "alpha": 0.1}),
+     (Ridge, {"solver": 'sparse_cg', 'tol': 1e-12, "alpha": 0.1}),
+     (LinearRegression, {}),
+     (RidgeCV, {})
+     ])
+def test_model_pipeline_same_dense_and_sparse(LinearModel, params):
+    # Test that linear model preceeded by StandardScaler in the pipeline and
+    # with normalize set to False gives the same y_pred and the same .coef_
+    # given X sparse or dense
+
+    rng = np.random.RandomState(0)
+    n_samples = 200
+    n_features = 2
+    X = rng.randn(n_samples, n_features)
+    X[X < 0.1] = 0.
+
+    X_sparse = sparse.csr_matrix(X)
+    y = rng.rand(n_samples)
+
+    if is_classifier(LinearModel):
```

Comment:
Either move the model instantiation before or use issubclass as in the previous test.